### ISSUE TITLE
mount_paths is not working with --shell

### ISSUE THREAD

**Issue by luogni** [stepchowfun/toast#261]
If I use mount_paths and start toast with --shell I get empty files and directories. With this toast file:
```
image: ubuntu:16.04
default: mount
tasks:
  mount:
    cache: false
    mount_paths:
      - foo
    command: |
      cat foo
```
and:
```
echo "test" > foo
```
I get this problem:
```
luogni@luogni-XPS-13-9350:~/Temp/a$ toast
[INFO] Ready to run 1 task: mount.
[INFO] Running task mount…
test
luogni@luogni-XPS-13-9350:~/Temp/a$ toast --shell
[INFO] Ready to run 1 task: mount.
[INFO] Running task mount…
test
[INFO] Preparing a shell…
root@efe4bc98ca20:/scratch# ls -al
total 8
drwxrwxrwx 2 root root 4096 Jan  1  1970 .
drwxr-xr-x 1 root root 4096 Jun 20 12:20 ..
-rwxr-xr-x 1 root root    0 Jun 20 12:20 foo
root@efe4bc98ca20:/scratch# cat foo
root@efe4bc98ca20:/scratch# 
```

**Comment by stepchowfun** [comment]
Hi @luogni, this was actually the intended behavior. The `--shell` option takes effect _after_ the last task, not during it. So any paths that were mounted during the task's execution are no longer mounted when the interactive shell is started.

However, now I'm questioning whether this is the right design. The interactive shell does preserve some aspects of the final task: the environment variables, the location, and the user. It seems to me that we could be more consistent by including the `mount_path`s (and `mount_readonly` flag) and `ports` from the last task as well.

I'll take a look at this from an implementation standpoint soon. Thanks for bringing this to my attention!

**Comment by luogni** [comment]
Ok thanks, this is my goal:
*  define test task that build software and run unit testing
*  if this fails I can run it with --shell and hack until it works again

**Comment by stepchowfun** [comment]
Hi @luogni, I believe this is now fixed by https://github.com/stepchowfun/toast/pull/264. The fix was released in version 0.28. You can upgrade to that version and let me know if you run into any issues. Thanks again for reporting this!
